packages/astro/src/runtime/server/render/component.ts

```diff
@@ -128,23 +128,31 @@ async function renderFrameworkComponent(
 		}
 
 		if (!renderer) {
-			let error;
-			for (const r of renderers) {
-				try {
-					if (await r.ssr.check.call({ result }, Component, props, children)) {
-						renderer = r;
-						break;
+ 			// If there's only one renderer in the project (solid-js)
+			// we can skip the `check` calls and use that renderer (to not render solid component twice)
+			if (renderers.length === 1 && renderers[0].name === '@astrojs/solid') {
```

Comment:
Why only Solidjs?

Reply:
because for some reason it fails for other frameworks. Moreover for solid it's critical to be rendered once, because solid doesn't have this "rerender" paradigm as react, preact or others.

More common solution could require much more efforts, so it should be carefully implemented. This is really easy fix which can help many people, so we can do improvements step by step.

Reply:
I tried to apply the solution suggested [here](https://github.com/withastro/astro/issues/8368), but looks like it doesn't work anymore for all the frameworks.

Reply:
What tests fail? It's likely something we can fix. I don't think we should special-case one particular renderer in this case. There's nothing about this suggestion that shouldn't work everywhere.

If you could post the errors that occur when this is enabled for all renderers that would be helpful.

Reply:
I would appreciate if you help to understand why the tests are failing.

Reply:
Just pushed a commit

Reply:
thanks, will take a look.

Reply:
maybe here we need to check not `renderers` but `validRenderers`? I see that there can be a renderer with `astro:jsx` name, maybe this is the point...

Reply:
No, the problem is that some renderers (Preact in particular), expect both `check` and `renderToStaticMarkup` to be called because they store state in `check`. We can probably make it a breaking change in 5.0 that `check` is not guaranteed to always be called.

Easy fix though, pushed a change.

Reply:
Well there must be another issue too, will look into that.